js/platform.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import * as fbs from "gen/msg_generated";
+import { flatbuffers } from "flatbuffers";
+import { assert } from "./util";
+import * as dispatch from "./dispatch";
+
+export interface PlatformInfo {
+  os: string;
+  family: string;
+}
+
+let cachedPlatformInfo: PlatformInfo | null = null;
+
+/**
+ * Retrieves information (os, os family) about the current
+ * platform synchronously.
+ *
+ *     import { platformSync } from "deno";
+ *     const platform = deno.platformSync();
+ */
+export function platformSync(): PlatformInfo {
+  if (!cachedPlatformInfo) {
+    cachedPlatformInfo = Object.freeze(res(dispatch.sendSync(...req())));
+  }
+  return cachedPlatformInfo!;
+}
```

Comment:
I don’t think we need an async version..?

Reply:
I was originally thinking about making the format uniform for each call (do agree async does not really create obvious benefits), but since we are not having an async version for `deno.env()` I should probably just drop this one.